### PR TITLE
Parse transit encoded listing images

### PR DESCRIPTION
**Pull request by mporkola**

### CODE REVIEW

File: client/app/startup/SearchPageApp.js [hunk 1]
```diff
@@ -21,24 +23,18 @@ export default (props) => {
...
-  //       title: l.get(':attributes').get(':title'),
-  //     }))
-  //     .toSet(),
-  // });
-
-  const combinedProps = Object.assign({}, { searchPage: new SearchPageModel() }, { routes });
+  const bootstrappedData = TransitImmutableConverter.fromJSON(props.data);
+  const searchPage = new SearchPageModel({
+    currentPage: bootstrappedData.get(':data')
+      .get(0)
```
Comment (mporkola) [inline client/app/startup/SearchPageApp.js]: Related to a bug in backend implementation, will be removed before merge

File: client/app/startup/SearchPageApp.js [hunk 2]
```diff
@@ -21,24 +23,18 @@ export default (props) => {
...
+      .map((l) => l.get(':id')),
+    listings: bootstrappedData
+      .get(':data')
+      .get(0)
```
Comment (mporkola) [inline client/app/startup/SearchPageApp.js]: same here

File: client/app/utils/transitImmutableConverter.js [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
+    ':square': 'square',
+    ':square_2x': 'square2x',
+  };
+  const data = rep[1];
+  const images = data.map((s) => new Image({
+    type: s.get(0),
+    height: s.get(1),
+    width: s.get(2), // eslint-disable-line no-magic-numbers
+    url: s.get(3), // eslint-disable-line no-magic-numbers
+  }));
```
Comment (Gnito) [inline client/app/utils/transitImmutableConverter.js]: Alternatively:
const images = data.map(([type, height, width, url] = s) => new Image({ type, height, width, url }));

File: client/app/components/sections/SearchPage/SearchPage.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ class SearchPage extends Component {
 
   constructor(props, context) {
     super(props, context);
-    this.listings = [];
+    this.listings = props.searchPage.listings || [];
```
Comment (Gnito) [inline client/app/components/sections/SearchPage/SearchPage.js]: This should be currentListings in the future. (Listings filtered by currentPage indexes)

File: client/app/startup/SearchPageApp.js [hunk 5]
```diff
@@ -24,14 +26,22 @@ export default (props) => {
...
+      const listing = parseListingModel(val);
+      return acc.set(listing.id, listing);
+    }, new Immutable.Map());
+
+  const rawListings = bootstrappedData
+    .get(':data')
+    .get(0);
+
   const searchPage = new SearchPageModel({
     currentPage: bootstrappedData.get(':data')
```
Comment (Gnito) [inline client/app/startup/SearchPageApp.js]:
```javascript
currentPage: rawListings
  .map((l) => l.get(':id')),
```

File: client/app/utils/transitImmutableConverter.js [hunk 6]
```diff
@@ -1,6 +1,33 @@
...
+  square: new Image(),
+  square2x: new Image(),
+});
+
+const toImage = (rep) => {
+  const knownStyles = {
+    ':square': 'square',
+    ':square_2x': 'square2x',
+  };
+  const data = rep[1];
```
Comment (Gnito) [inline client/app/utils/transitImmutableConverter.js]: I had to remove this in my branch, but it should probably be done here instead:
https://github.com/sharetribe/sharetribe/pull/2582/files?diff=unified#diff-17b1cd2a70336c85378e752ce85a7bd9L16

**Pull request merged**